Remove time from event startDate / endDate fields in the CMS [skip percy]

CharlyMartin opened the pull request:
> [!NOTE]
> This is PR 1/3 aimed at improving the date UX in the CMS. It will be merged last once PR 2 and 3 are merged into this one.

## 📝 Description

This PR updates the event's `startDate` / `endDate` fields by removing the time portion. This will improve the CMS UX for the content team, standardize the data and make things clearer for us.

## 🛠️ Key Changes

- Updates `config.yml` with new datetime format - https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/1091/commits/7a611725156f4bda4b914b095922982a3519f502
- Removes the time portion from all event Markdown entries - https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/1091/commits/06fd2ec27c39393a746780bbd95dd5beda32c201
- Refactors `getMetaData` to display a single date if the start and end dates are the same.
- Fixes a bug in event sorting where an event can be both upcoming and past, as seen on [fil.org](https://fil.org/events) today with the _Digital Real Estate Summit_ event - https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/1091/commits/7920f0cf9aa0e8a18508d90d750e6b84db8fbf27

## 🧪 How to Test

Run the CMS locally, create an event and make sure the start and end times are only dates, not times.

## 📸 Screenshots

![CleanShot 2025-02-12 at 14 31 57@2x](https://github.com/user-attachments/assets/c5ea3259-e3d2-4fd6-b70a-0ecadd02f071)

## 🔖 Resources

- https://decapcms.org/docs/widgets/datetime/

barbaraperic commented:
> Just testing for now! I've added end date, and removed it - which caused a zod error:
> 
> ```
> Error [ZodError]: [
>   {
>     "code": "invalid_date",
>     "path": [
>       "end-date"
>     ],
>     "message": "Invalid date"
>   }
> ]
> ```
> 
> Basically the field went from `end-date: 2025-02-13` to `end-date: ""`

barbaraperic commented:
Think it's unrelated to this PR, but I was able to create an event without SEO description, and zod started screaming.


```
 ⨯ Error [ZodError]: [
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "undefined",
    "path": [
      "seo",
      "description"
    ],
    "message": "Required"
  },
  {
    "code": "invalid_date",
    "path": [
      "end-date"
    ],
    "message": "Invalid date"
  }
]
    at async getMarkdownData (src/app/_utils/getMarkdownData.ts:39:23)
    at async getEventsData (src/app/events/utils/getEventData.ts:17:20)
    at async Object.generateStaticParams (webpack-internal:/(rsc)/src/app/events/[slug]/src/app/events/[slug]/page.tsx:137:18)
  37 |     const dataToValidate = content ? { ...data, content } : data
  38 |
> 39 |     const parsedData = await zodSchema.parseAsync(dataToValidate)
     |                       ^

```

barbaraperic commented:
Suuuper random edge case, but we allow start-date to be after end-date. Example: start-date: 2025-02-12
end-date: 2025-02-13.

barbaraperic commented on apps/site/src/app/events/utils/getMetaData.ts:
```diff
@@ -14,8 +16,17 @@ export function getMetaData({
-  const formattedDate = `${formattedStartDate}${formattedEndDate}`
 
-  return [formattedDate, location]
+  if (!endDate) {
+    return [formattedStartDate, location]
+  }
+
+  if (isSameDay(startDate, endDate)) {
+    return [formattedStartDate, location]
+  }
+
+  const formattedEndDate = formatDate(endDate)
+  const dateRange = `${formattedStartDate} - ${formattedEndDate}`
+
+  return [dateRange, location]
```
this reads nicely 😊

mirhamasala commented:
>Think it's unrelated to this PR, but I was able to create an event without SEO description, and zod started screaming.

@barbaraperic  This has been fixed in https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/1098

mirhamasala commented on apps/site/src/app/events/utils/getMetaData.ts:
```diff
@@ -1,10 +1,12 @@
 type getMetaDataParams = {
   startDate: Event['startDate']
-  endDate?: Event['endDate']
+  endDate: Event['endDate']
```
Why is this not optional anymore?

CharlyMartin replied:
`endDate` is `Date | undefined` so it's still optional.

We can't use `Pick<Event, ...>` because `location` is nested in the object: `Event['location']['primary']`

mirhamasala commented on apps/site/src/content/events/agi-cryptography-security-multipolar-scenarios-workshop.md:
```diff
@@ -7,8 +7,8 @@ category: supported-sponsored
   region: north-america
-start-date: 2024-05-14T15:48:19.518Z
-end-date: 2024-05-15T15:48:19.530Z
+start-date: 2024-05-14
```
Have we tested this using Chrome DevTools → Network tab → More tools → Sensors?

I’m concerned that some systems (browsers) might assume the date is in local time, especially if they automatically attach a time zone.

CharlyMartin replied:
That's a valid concern!

When logging the data here I see this, which is what we want!

`startDate 2024-11-06T00:00:00.000Z`
`endDate 2024-11-12T00:00:00.000Z`

But I'll play with the Sensors and see if we get different results in the US vs Asia.

CharlyMartin replied:
I set my timezone as New Zealand and the date format was the same as here: `2024-11-06T00:00:00.000Z`

CharlyMartin commented:
> Just testing for now! I've added end date, and removed it - which caused a zod error:

Yes, this is not related to this PR, this how Decap works. It doesn't delete fields 🥲

CharlyMartin commented on apps/site/src/content/events/dweb-camp.md:
```diff
@@ -8,8 +8,8 @@ location:
 external-link: https://dwebcamp.org/
-start-date: 2022-08-24T11:30:00.000Z
-end-date: 2022-08-22T11:30:00.000Z
+start-date: 2022-08-24
+end-date: 2022-08-22
```
It should definitely be the other way around 😅 And that's a good catch! Nothing prevents the author from doing this, let's add a Zod validation to catch these for us 🙏

CharlyMartin replied:
https://fil.org/events/dweb-camp 🙃

CharlyMartin replied:
https://www.notion.so/filecoin/FF-Look-into-CMS-quirks-b1bea7e42c16455baf22752cb4e3aa9e?pvs=4#19d7631f282580e9912ce32b8fb7e1ee

mirhamasala commented on apps/site/src/app/events/schemas/EventBaseFrontmatterSchema.ts:
```diff
@@ -1,9 +1,11 @@
+export const EventBaseFrontmatterSchema = z
+  .object({
+    title: z.string(),
+    'start-date': z.date(),
```
What's the reasoning behind removing `coerce`?

mirhamasala commented on apps/site/src/content/events/consensus-filecoin-orbit-event.md:
```diff
@@ -10,8 +10,8 @@ location:
-start-date: 2024-05-30T16:00:00.000Z
-end-date: 2024-05-30T17:00:00.000Z
+start-date: 2024-05-30
+end-date: 2024-05-30
```
```suggestion
```

mirhamasala commented on apps/site/src/content/events/espa-bootcamp-cohort-4.md:
```diff
@@ -8,8 +8,8 @@ location:
-start-date: 2022-10-13T00:00:00.000Z
-end-date: 2022-10-13T00:00:00.000Z
+start-date: 2022-10-13
+end-date: 2022-10-13
```
```suggestion
```

mirhamasala commented on apps/site/src/content/events/filecoin-orbit-kampala-fvm-hack-day.md:
```diff
@@ -10,8 +10,8 @@ category: supported-sponsored
-start-date: 2024-12-14T09:00:00.000Z
-end-date: 2024-12-14T18:00:00.000Z
+start-date: 2024-12-14
+end-date: 2024-12-14
```
```suggestion
```

mirhamasala commented on apps/site/src/content/events/filecoin-orbit-playground-16-ai-and-data-storage-for-workforce.md:
```diff
@@ -13,8 +13,8 @@ category: supported-sponsored
-start-date: 2024-12-23T17:00:00.000Z
-end-date: 2024-12-23T20:00:00.000Z
+start-date: 2024-12-23
+end-date: 2024-12-23
```
```suggestion
```

mirhamasala commented on apps/site/src/content/events/sbs-boston.md:
```diff
@@ -7,8 +7,8 @@ category: supported-sponsored
-start-date: 2023-04-13T15:36:02.820Z
-end-date: 2023-04-13T15:36:00.000Z
+start-date: 2023-04-13
+end-date: 2023-04-13
```
```suggestion
```

mirhamasala commented on apps/site/src/content/events/singapore-storage-provider-meetup.md:
```diff
@@ -7,8 +7,8 @@ location:
-start-date: 2022-08-16T00:00:00.000Z
-end-date: 2022-08-16T00:00:00.000Z
+start-date: 2022-08-16
+end-date: 2022-08-16
```
```suggestion
```

mirhamasala reviewed (approved):
@CharlyMartin - Happy to get this merged! Left a questions and some suggestions that can be directly committed from GitHub. Thank you! 🙌🏼

CharlyMartin commented:
> @CharlyMartin - Happy to get this merged! Left a questions and some suggestions that can be directly committed from GitHub. Thank you! 🙌🏼

Thanks Mirha!

I'll keep this open until we merge 2 other PRs in this one:
- One [PR](https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/1096/files) to remove the time segment from `date`
- And one to remove it from `program` > `start-date` & `end-date`

So that everything is in place at once for the content team!

mirhamasala reviewed (approved):
LGTM! ❤️
